### PR TITLE
fix colspan and validation for spanning columns

### PR DESCRIPTION
**Pull request by nburka**

### CODE REVIEW

**Comment by nburka** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;uiZ3lxrx&#x2F;46-d-allow-for-submission-with-pending-sis"> ½d - allow for submission with pending SIs</a>

**Comment by charleswaddell** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;PBcMvNQ9&#x2F;47-1d-marketing-total-issues"> 1d - marketing total issues</a>

File: Swat/SwatTableView.php [hunk 1]
```diff
@@ -412,14 +412,27 @@ public function getMessages()
...
+					$messages = array_merge(
+						$messages,
+						$column->getMessages($row)
+					);
+				}
+			}
+
+			foreach ($this->extra_rows as $row) {
+				if ($row instanceof SwatTableViewWidgetRow) {
 					$messages =	array_merge($messages, $row->getMessages());
```
Comment (gauthierm) [inline Swat/SwatTableView.php]: ws (tab should be a space)

File: Swat/SwatTableView.php [hunk 2]
```diff
@@ -474,8 +494,13 @@ public function hasMessage()
 	public function getXhtmlColspan()
 	{
 		$count = 0;
-		foreach ($this->getVisibleColumns() as $column)
+		foreach ($this->getVisibleColumns() as $column) {
 			$count += $column->getXhtmlColspan();
+		}
+
+		if ($count < 2 && count($this->getSpanningColumns()) > 0) {
```
Comment (gauthierm) [inline Swat/SwatTableView.php]: Why is this lower limit of 2 needed? Shouldn't it be a minimum of 1?

Comment (nburka) [inline Swat/SwatTableView.php]: a spanning column has two cols - the title and the contents.

Comment (gauthierm) [inline Swat/SwatTableView.php]: The title for a spanning column is optional. If it is not set only one column is used.

Comment (nburka) [inline Swat/SwatTableView.php]: @gauthierm what I have is a TableView with only spanning columns a group-by field. The way it currently is, the group-by header only spans one column and is squished. So we either need to:
 - check if any spanning-column has a title and set to `2`
 - not use a table-view that way, and instead use a replicator

Comment (gauthierm) [inline Swat/SwatTableView.php]: See https://github.com/nburka/swat/pull/2

**Pull request merged**